src/os/bluestore/BlueFS.h

```diff
@@ -558,16 +560,19 @@ class BlueFS {
   void handle_discard(unsigned dev, interval_set<uint64_t>& to_release);
 
   void flush(FileWriter *h) {
-    std::lock_guard l(lock);
+    std::unique_lock l(lock);
     _flush(h, false);
+    _maybe_compact_log(l);
```

Comment:
This looks like an overkill to me -IIUC flush doesn't guarantee data persistence and hence KV has to use FSync from time to time. I.e. compacting log from fsync is likely to be enough. Besides I'm curious about performance impact of this call on each flush.

Reply:
@ifed01 Modified to check only when flush only really flushed something to disk

Reply:
@aclamk - but I'm still curious if compacting log on fsync only would be enough? Does RocksDB call fsync often enough during that infrequent writes processing which you're referring in the description.

Reply:
@ifed01 Customer had a cluster where there was almost no write traffic. Only op was write to WAL for 23 bytes once every second. Unfortunately, it translated to 109 kB of BlueFS log per second (mostly due to BlueFS log containing >70000 extents).